xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/CollectionNode.java

```diff
@@ -60,14 +68,23 @@
      */
     private int maxLeafNodes = -1;
 
-    public CollectionNode(PubSubService service, CollectionNode parentNode, String nodeID, JID creator) {
-        super(service, parentNode, nodeID, creator);
-        // Configure node with default values (get them from the pubsub service)
-        DefaultNodeConfiguration defaultConfiguration = service.getDefaultNodeConfiguration(false);
+    public CollectionNode( PubSubService.UniqueIdentifier serviceId, CollectionNode parentNode, String nodeID, JID creator, boolean subscriptionEnabled, boolean deliverPayloads, boolean notifyConfigChanges, boolean notifyDelete, boolean notifyRetract, boolean presenceBasedDelivery, AccessModel accessModel, PublisherModel publisherModel, String language, ItemReplyPolicy replyPolicy, LeafNodeAssociationPolicy associationPolicy, int maxLeafNodes)
```

Comment:
Ouch - that's a lot of constructor params and could rapidly become unmaintainable. I think this is possibly a good case for either a configuration POJO as a single constructor param or a builder pattern.

FWIW, I've become a great fan of Project lombok for this sort of thing.
```
@Builder
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
```
Will create a class that can only be constructed with a builder API, and create the builder too.

Reply:
I've introduced this monstrosity to avoid the default config to be retrieved in the constructor (thus decoupling responsibilities).

There is an overloaded variant of this that takes a DefaultNodeConfiguration pojo - which basically does what you want. However, that pojo is re-used for both types of Nodes, which both require a different subset from its configuration.

We might want to split up that DefaultNodeConfiguration implementation in more specific variants.